### ISSUE TITLE
Outbounds should block until they are started

### ISSUE THREAD

**Issue by willhug** [yarpc/yarpc-go#721]
Currently if an outbound is called and it has not yet been started, we will return an error that says it has not started.  Instead, we should block until the outbound is started, our the timeout of the request is exceeded.

This is now an issue with TChannel inbounds since incoming requests could hit outbounds before they have been started (since TChannel will start taking requests as soon as the Transport is started (which is before outbounds))

**Comment by willhug** [comment]
We could probably build this into lifecycleOnce to have a channel that can be listened on

**Comment by kriskowal** [comment]
We have addressed this, but I worry that we have solved it the wrong way.

It may be better to alter the lifecycle contract. Proposal: If you construct a lifecycle, you are obliged to call Stop(), regardless of whether you call Start(). Peer lists may be updated inside or outside the lifecycle. For every addition, the peer list must accept a corresponding removal.

The last assertion is broken because the peer list depopulates itself on Stop, so an update after Stop produces an error.

Requiring Stop() regardless of Start() might not be necessary. It could be that the peer list buffers up identifiers it's been introduced to and retains them on Start(). Retaining before Start() is why we need to release them on Stop()
